tests/pl/test_render_points.py

```diff
@@ -225,6 +226,72 @@ def test_plot_datashader_can_transform_points(self, sdata_blobs: SpatialData):
 
         sdata_blobs.pl.render_points("blobs_points", method="datashader", color="black", size=5).pl.show()
 
+    def test_plot_can_use_norm_with_clip(self, sdata_blobs: SpatialData):
+        cmap = matplotlib.colormaps["viridis"]
+        cmap.set_under("black")
+        cmap.set_over("grey")
+        sdata_blobs.pl.render_points(color="instance_id", size=40, norm=Normalize(3, 7, clip=True), cmap=cmap).pl.show()
+
+    def test_plot_can_use_norm_without_clip(self, sdata_blobs: SpatialData):
+        cmap = matplotlib.colormaps["viridis"]
+        cmap.set_under("black")
+        cmap.set_over("grey")
+        sdata_blobs.pl.render_points(
+            color="instance_id", size=40, norm=Normalize(3, 7, clip=False), cmap=cmap
+        ).pl.show()
+
+    def test_plot_datashader_can_use_norm_with_clip(self, sdata_blobs: SpatialData):
+        cmap = matplotlib.colormaps["viridis"]
+        cmap.set_under("black")
+        cmap.set_over("grey")
+        sdata_blobs.pl.render_points(
+            color="instance_id",
+            size=40,
+            norm=Normalize(3, 7, clip=True),
+            cmap=cmap,
+            method="datashader",
+            datashader_reduction="max",
+        ).pl.show()
+
+    def test_plot_datashader_can_use_norm_without_clip(self, sdata_blobs: SpatialData):
+        cmap = matplotlib.colormaps["viridis"]
+        cmap.set_under("black")
+        cmap.set_over("grey")
+        sdata_blobs.pl.render_points(
+            color="instance_id",
+            size=40,
+            norm=Normalize(3, 7, clip=False),
+            cmap=cmap,
+            method="datashader",
+            datashader_reduction="max",
+        ).pl.show()
+
+    def test_plot_datashader_norm_vmin_eq_vmax_with_clip(self, sdata_blobs: SpatialData):
+        cmap = matplotlib.colormaps["viridis"]
+        cmap.set_under("black")
+        cmap.set_over("grey")
+        sdata_blobs.pl.render_points(
+            color="instance_id",
+            size=40,
+            norm=Normalize(5, 5, clip=True),
+            cmap=cmap,
+            method="datashader",
+            datashader_reduction="max",
+        ).pl.show()
+
+    def test_plot_datashader_norm_vmin_eq_vmax_without_clip(self, sdata_blobs: SpatialData):
+        cmap = matplotlib.colormaps["viridis"]
+        cmap.set_under("black")
```

Comment:
you always create the same cmap here. Please make some primitive helper function like `_get_cmap_with_under_over("viridis", "black", "grey")` which you can then also reuse in your conftest